src/localplayer.cpp

```diff
@@ -586,31 +587,31 @@ void LocalPlayer::applyControl(float dtime, Environment *env)
 	}
 
 	if (continuous_forward)
-		speedH += move_direction;
+		speedH += v3f(0,0,1);
 
 	if (control.up) {
 		if (continuous_forward) {
 			if (fast_move)
 				superspeed = true;
 		} else {
-			speedH += move_direction;
+			speedH += v3f(0,0,1);
 		}
 	}
 	if (control.down) {
-		speedH -= move_direction;
+		speedH -= v3f(0,0,1);
 	}
 	if (!control.up && !control.down) {
-		speedH -= move_direction *
+		speedH -= v3f(0,0,1) *
 			(control.forw_move_joystick_axis / 32767.f);
 	}
 	if (control.left) {
-		speedH += move_direction.crossProduct(v3f(0,1,0));
+		speedH += v3f(-1,0,0);
 	}
 	if (control.right) {
-		speedH += move_direction.crossProduct(v3f(0,-1,0));
+		speedH += v3f(1,0,0);
 	}
 	if (!control.left && !control.right) {
-		speedH -= move_direction.crossProduct(v3f(0,1,0)) *
+		speedH -= v3f(1,0,0) *
```

Comment:
Just to make sure: Is this line correct? Looking at L608 (and the old code), shouldn't this be
`speedH -= v3f(-1,0,0) * ....` or `speedH += v3f(1,0,0) * ....`?

Reply:
You are right. I did a mistake in my cross product. But this needs testing and I don't have the material for that.